app/assets/javascripts/components/govspeak-editor.js

```diff
@@ -47,63 +79,94 @@ window.GOVUK.Modules = window.GOVUK.Modules || {}
     return data
   }
 
-  GovspeakEditor.prototype.initPreview = function () {
-    const previewToggle = this.module.querySelector(
-      '.js-app-c-govspeak-editor__preview-button'
+  GovspeakEditor.prototype.maybeTrackEvent = function (label) {
+    if (this.shouldTrackToggle) {
+      GOVUK.analytics.trackEvent(
+        this.previewButton.getAttribute('data-preview-toggle-track-category'),
+        this.previewButton.getAttribute('data-preview-toggle-track-action'),
+        { label }
+      )
+    }
+  }
+
+  GovspeakEditor.prototype.showPreview = function (event) {
+    event.preventDefault()
+
+    this.backButton.classList.add('app-c-govspeak-editor__back-button--show')
+    this.previewButton.classList.remove(
+      'app-c-govspeak-editor__preview-button--show'
```

Comment:
Would storing these class names in a variable, e.g. `showPreviewButtonClass`, make it any easier to read what's going on? I'm finding it tough to follow all the BEM class names, as you suggested I would

Reply:
I think that I want to follow up with another PR on this, potentially introducing a slimmed down version of i-bem.js to whitehall or adding some context specific functions to this module for toggling visibility.